e2e_tests/tests/all_providers/normal.rs

```diff
@@ -147,3 +151,19 @@ fn list_keys() {
     assert!(key_names.contains(&(key2.clone(), ProviderID::Pkcs11)));
     assert!(key_names.contains(&(key3.clone(), ProviderID::Tpm)));
 }
+
+#[test]
+// See #310
+fn invalid_provider_list_keys() {
+    let mut client = RawRequestClient {};
+    let mut req_hdr = RawHeader::new();
+
+    // Always targeting the Mbed Crypto provider
+    req_hdr.provider = 0x1;
```

Comment:
Do you think we should try and loop here as well, to test all the providers?

Reply:
Ideally yes I guess. The thing is that it might be a bit over the top for this test to first spawn a normal test client to get all provider IDs and then use the raw client to test all provider IDs.